calico/common.py

```diff
@@ -547,3 +552,41 @@ def validate_tags(profile_id, tags):
 
     if issues:
         raise ValidationFailed(" ".join(issues))
+
+def validate_ipam_pool(pool_id, pool, ip_version):
+    # Remove any keys that we're not expecting.  Stops unvalidated data from
+    # slipping through.  We ignore other keys since this structure is used
+    # by calicoctl for its own purposes too.
+    keys_to_remove = set()
+    for key in pool:
+        if key not in ("cidr", "ipip", "masquerade"):
```

Comment:
Nothing wrong with this, but we should move this list to a variable with a helpful name. It's a bit magic where it is now.